src/test/scala/com/codacy/analysis/cli/tools/ToolSpec.scala

```diff
@@ -105,13 +109,57 @@ class ToolSpec extends Specification with NoLanguageFeatures {
           Option.empty,
           Option(Map(Languages.Java -> LanguageConfiguration(Option(Set("rawr")))))))
 
-      val toolEither = AnalyseExecutor.tools(userInput, localConfiguration, remoteProjectConfiguration, filesTarget)
+      val toolEither = AnalyseExecutor
+        .tools(userInput, localConfiguration, remoteProjectConfiguration, filesTarget, allowNetwork = true)
       toolEither must beRight
       toolEither must beLike {
         case Right(toolSet) =>
           toolSet.map(_.name) mustEqual Set("checkstyle", "findbugs", "findbugssec", "pmd")
       }
     }
-  }
 
+    """return an informative error message if the user selects a tool that needs access
+      |to the network but doesn't provide the needed argument""".stripMargin in {
+      val toolName = "gendarme"
+      val remoteProjectConfiguration = Left("some error")
+      val filesTarget = FilesTarget(File(""), Set(File("Test.cs").path))
+      val localConfiguration = Right(CodacyConfigurationFile(Option.empty, Option.empty, Option.empty))
+
+      val toolEither = AnalyseExecutor
+        .tools(Some(toolName), localConfiguration, remoteProjectConfiguration, filesTarget, allowNetwork = false)
+
+      toolEither must beLeft(
+        s"The tool $toolName needs network access to execute. Run the command with the parameter 'allow-network', or '-n'.")
```

Comment:
Food for thought: in the future we could consider moving all the error messages somewhere so we don't replicate them in tests and have a nice file with them aggregated.

Reply:
yes, when I was writing this test something like that also came to my mind...